src/main/java/nell/Nell.java

```diff
@@ -0,0 +1,346 @@
+     */
+    private static void markTask(String taskNumber) {
+        try {
+            int taskIndex = Integer.parseInt(taskNumber);
+            tasks.getTaskAtIndex(taskIndex - 1).setDone(true);
+            System.out.println("-> The following task has been marked done:");
+            printTaskAtIndex(tasks.getTaskAtIndex(taskIndex - 1), taskIndex);
+        } catch (NumberFormatException e) {
+            System.out.print(MARK_ERROR_MESSAGE);
+        } catch (IndexOutOfBoundsException e) {
+            System.out.println(INVALID_TASK_MESSAGE);
+        }
+    }
+
+    /**
+     * Removes a task from the list
+     *
+     * @param taskNumber The command body
+     */
+    private static void removeTask(String taskNumber) {
+        try {
+            int taskIndex = Integer.parseInt(taskNumber);
+            tasks.removeTask(taskIndex - 1);
+        } catch (NumberFormatException e) {
+            System.out.print(REMOVE_ERROR_MESSAGE);
+        } catch (IndexOutOfBoundsException e) {
+            System.out.println(INVALID_TASK_MESSAGE);
+        }
+    }
+
+    /**
+     * Handles wrong or wrongly formatted commands
+     */
+    private static void handleIncorrectInput() {
+        System.out.print("""
+                -> Invalid command!
+                   Please enter one of the following commands:
+                      list
+                      mark <number>
+                      unmark <number>
+                      todo <description>
+                      deadline <description> /by <by-date>
+                      event <description> /from <from-date> /to <to-date>
+                      remove <number>
+                      bye
+                """);
+    }
+
+    /**
+     * Says bye to the user
+     */
+    private static void sayBye() {
+        System.out.println("-> Bye. Hope to see you again soon!");
+    }
+
+    private static void loadFromFile() {
+        try {
+            File directory = new File("data");
+            File dataFile = new File("data/data.txt");
+
+            if (!dataFile.exists()) {
+                directory.mkdir();
+                dataFile.createNewFile();
+            }
+
+            Scanner fileScanner = new Scanner(dataFile);
+            while (fileScanner.hasNext()) {
+                String fileTask = fileScanner.nextLine();
+                loadTask(fileTask);
+            }
+        } catch (IOException e) {
+            System.out.println("Data not loaded - File not found");
+        }
+    }
+
+    /**
+     * Parses a task from the file
+     *
+     * @param taskLine The file line containing the task
+     */
+    private static void loadTask(String taskLine) {
+        String[] taskParameters = taskLine.split("\\|");
+        String taskType = taskParameters[0];
+        String taskDescription = taskParameters[2];
+
+        boolean taskIsDone = false;
+        if (taskParameters[1].equals("X"))
+        {
+            taskIsDone = true;
+        }
```

Comment:
Could be simplified to one line: boolean taskIsDone = taskParameters[1].equals("X");